ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/AbstractPartitionPlan.java

```diff
@@ -277,20 +312,30 @@ protected class BinPackingTaskSplitter implements TaskSplitter {
     @Override
     public List<SplitTask> splitTasks(int targetTaskCount) {
       // bin-packing
-      List<FileTask> allDataFiles = Lists.newArrayList();
+      List<FileTask> fileTasks = Lists.newArrayList();
       rewriteDataFiles.forEach(
-          (dataFile, deleteFiles) -> allDataFiles.add(new FileTask(dataFile, deleteFiles, true)));
+          (dataFile, deleteFiles) -> fileTasks.add(new FileTask(dataFile, deleteFiles, true)));
       rewritePosDataFiles.forEach(
-          (dataFile, deleteFiles) -> allDataFiles.add(new FileTask(dataFile, deleteFiles, false)));
+          (dataFile, deleteFiles) -> fileTasks.add(new FileTask(dataFile, deleteFiles, false)));
+      List<SplitTask> results = Lists.newArrayList();
+      results.addAll(genSplitTasks(fileTasks));
+
+      fileTasks.clear();
+      rewriteSegmentDataFiles.forEach(
```

Comment:
Doing bin-packing together with fragment datafile will cause write amplification problems. If each checkpoint generates a small datafile, it will merge a file close to 96m, and each merge will trigger rewriting.